packages/desktop-client/src/components/common/Menu.tsx

```diff
@@ -32,6 +33,9 @@ type MenuItem = {
   text: string;
   key?: string;
   style?: CSSProperties;
+  toggle?: boolean;
```

Comment:
Would it work if we introduce a new `TogglableMenuItem` which extends from `MenuItem`?

Reply:
This only works if we are looking to group toggled items and non-toggled items. If you want a menu with (regular item, toggle item, regular item, toggle item) then a separate toggleitem group would not work without an overly complicated code set. I know we have a couple complex menus right now. Not sure there's any that fit this use case currently but I can imagine we may want this ability in the future. 

Thoughts?

Reply:
@joel-jeremy I'd love to have you thoughts on grouping toggle items at top or bottom vs my proposal which would allow for them to be used anywhere in the list. Cheers!

Reply:
I think it wouldn't hurt to allow the consumer of the component to place the toggle/regular item wherever it sees fit. It could group the toggles together or not, the component provides the flexibility. :)

Reply:
Hey @joel-jeremy, I've sent you a couple messages via discord. I don't see a way to implement how you are suggesting. Can you give me some guidance? If you don't have time to write anything right now, would you be okay with approving as is and we can adjust it to your vision in a new PR?